src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -600,7 +605,8 @@ public virtual string CreateToken(string payload, SigningCredentials signingCred
             if (encryptingCredentials == null && additionalHeaderClaims != null && additionalHeaderClaims.Count > 0)
                 header.Merge(JObject.FromObject(additionalHeaderClaims));
 
-            header.Merge(JObject.FromObject(AddCtyClaimDefaultValue(additionalInnerHeaderClaims)));
+            if (additionalInnerHeaderClaims != null && additionalInnerHeaderClaims.Count > 0)
```

Comment:
Similar to line 605, should we check if encryptcredentials is null and add inner headers if only so.